src/Utilities/functions.jl

```diff
@@ -202,7 +202,42 @@ coefficient(t::MOI.ScalarAffineTerm) = t.coefficient
 coefficient(t::MOI.VectorAffineTerm) = t.scalar_term.coefficient
 
 """
-    canonical(f::AbstractFunction)
+    copy(f::Union{ScalarAffineFunction, VectorAffineFunction})
+
+Return a new affine function with a shallow copy of the terms and constant(s)
+from `f`.
+"""
+Base.copy(f::F) where {F <: Union{SAF, VAF}} = F(copy(f.terms), copy(_constant(f)))
+
+"""
+    iscanonical(f::Union{ScalarAffineFunction, VectorAffineFunction})
+
+Returns a Bool indicating whether the function is in canonical form.
+See [`canonical`](@ref).
+"""
+function iscanonical(f::Union{SAF, VAF})
+    iscanonical(f.terms, termindices, t -> !iszero(coefficient(t)))
+end
+function iscanonical(x::AbstractVector, by, keep)
```

Comment:
It's not obvious from the signature the method does, could you add a docstring or (more informally) a comment?